### ISSUE TITLE
save freezes for some documents

### ISSUE THREAD

**Issue by shenkev** [mongoosastic/mongoosastic#274]
Hi,

I've been scratching my head over a peculiar problem I have and have tried everything. About 25% of my documents cannot be saved using the save function **when I run my web application on my cloud desktop**. However, **they save fine** when I try to do it on my **local machine**. 

I noticed that the document actually gets saved in MongoDB, it's only Elasticsearch that does not get the new entry (and after debugging, it seems to be the Elasticsearch indexing part of the save function that freezes).

Here is the schema:
var schema = {
  code               : { type: String, required: true, es_indexed: true },
  name              : { type: String, required: true, es_indexed: true, es_type: 'multi_field',
                        es_fields: {
                          name: { type: 'string', index: 'analyzed' },
                          rawname: { type: 'string', index: 'not_analyzed' }
                      }},
 subdocs        : { type:[subdoc], required: true, es_indexed: true }, // how do you avoid the _id that gets assigned to each entry in the list?
  description       : { type: String, required: true, es_indexed: true, es_type: 'multi_field',
                        es_fields: {
                          description: { type: 'string', index: 'analyzed' },
                          rawdescription: { type: 'string', index: 'not_analyzed' }
                      }},
  example           : { type: String, required: false, es_indexed: true },
  fact              : { type:String, required: false, es_indexed: true },
  links             : [{ type: String, required: false, es_indexed: true }],
  category          : { type: String, ref: 'Category', required: true, es_indexed: true },
  author            : { type:String, required: false, es_indexed: false }
};

Here is the error:

{ Error: Request Timeout after 30000ms
    at /home/ec2-user/Equationary/node_modules/elasticsearch/src/lib/transport.js:340:15
    at Timeout.<anonymous> (/home/ec2-user/Equationary/node_modules/elasticsearch/src/lib/transport.js:369:7)
    at ontimeout (timers.js:365:14)
    at tryOnTimeout (timers.js:237:5)
    at Timer.listOnTimeout (timers.js:207:5)
  status: 408,
  displayName: 'RequestTimeout',
  message: 'Request Timeout after 30000ms' }
2017-01-19T07:46:32.836+0000 W COMMAND  [conn2] the autoIndexId option is deprecated and will be removed in a future release

The log also mentioned something about mongoose promises being deprecated, I'm not sure if that's relevant:

(node:3573) DeprecationWarning: Mongoose: mpromise (mongoose's default promise library) is deprecated, plug in your own promise library instead: http://mongoosejs.com/docs/promises.html

**Comment by isayme** [comment]
seems like there net connection problem between your ES server and APP server.

**Comment by shenkev** [comment]
I am able to persist some of my documents successfully. This problem only occurs for a subset of documents, which makes me suspect it couldn't be a connection problem.

**Comment by shenkev** [comment]
After investigating further, I was able to resolve the issue by dropping the Elasticsearch index and Mongo index. I was able to successfully save all documents afterwards. I also noticed Elasticsearch behaving strangely. On Kibana I would search all documents in my index. Kibana displays hits = 16 but there are only 8 documents retrieved. 

This is kind of weird and suggests that Elasticsearch got corrupt?
